src/controllers/userService.js

```diff
@@ -0,0 +1,110 @@
+/* eslint-disable no-unused-vars */
+'use strict';
+
+let users = [];
+
+const init = () => {
+  users = [];
+};
+
+function findUserById(id) {
+  const numberId = Number(id);
+  const foundUser = users.find((user) => user.id === numberId);
+
+  return foundUser;
+}
+
+const createUser = (req, res) => {
+  const { name } = req.body;
+
+  if (!name) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const createdUser = {
+    id: Math.floor(Math.random() * 100),
+    name,
+  };
+
+  users.push(createdUser);
+  res.statusCode = 201;
+  res.send(createdUser);
+};
+
+const getAllUsers = (req, res) => {
+  res.statusCode = 200;
+  res.send(users);
+};
+
+const getUserById = (req, res) => {
+  const { id } = req.params;
+
+  const foundUsers = findUserById(id);
+
+  if (!foundUsers) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.statusCode = 200;
+
+  res.send(foundUsers);
+};
+
+const deleteUser = (req, res) => {
+  const { id } = req.params;
+
+  const foundUsers = findUserById(id);
+
+  if (!foundUsers) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  users = users.filter(user => user.id !== +id);
+  res.sendStatus(204);
+};
+
+const updateUser = (req, res) => {
+  const { id } = req.params;
+  const { name } = req.body;
+  let validationError = false;
+  const foundUsers = findUserById(id);
```

Comment:
findUserById returns just one user
```suggestion
  const foundUser = findUserById(id);
```